### ISSUE TITLE
Purge-like command?

### ISSUE THREAD

**Issue by yaneony** [rethinkdb/rethinkdb#6800]
After looking around the docs, i've noticed that there is .delete() command which should delete the data and, you can even delete all data in the table. The only problem is, it takes really long because it's deleting each row separately. Of cource, there is a way to drop and create table again, but that would mean you will also have to create/add all necessary secondary keys again, and it could be real pain on large tables.

Something like .purge() will be nice, which will kill all data within a table with single call.

**Comment by srh** [comment]
Yeah, that would be.  If we had this, we could actually implement this as an optimization of `r.table('foo').delete()`, instead of a separate command.  Or both.

There are some complexities in implementation when a replica has two shards on the same replica.

I think this also still has to interact with write hooks, if the table has a write hook that rejects deletions.  This is more of a user permissions issue -- purge could be used to bypass the constraint created by a write hook.  So maybe it's a feature that needs to be enabled in table config, or table write hook config, if the table has a write hook.

**Comment by gabor-boros** [comment]
Hm, I like the idea of having an improved delete or a dedicated purge function. I'm wondering what it would mean from the implementation point of view. Hm at the end I'll need to learn C++ to help Sam 😅

**Comment by gabor-boros** [comment]
Extremely old, but this is the same: https://github.com/rethinkdb/rethinkdb/issues/134

**Comment by yaneony** [comment]
> Dec 3, 2012
Extremely old :D But, for some case also was closed...

**Comment by marshallthomas** [comment]
Do you mind if I try to fix it?  I'm new to open source programming, but have done work with SQL database management code before based on JSON data mining.   The program I was working on would mine data about computer usage in a large tech building, and then upload it to a SQL database.  I had a similar problem and was able to fix it.  Granted, this was program was in python, but I have more experience with C++.

**Comment by srh** [comment]
Go for it!

**Comment by gabor-boros** [comment]
@marshallthomas If you need any help like how to run the tests, you can contact me anytime 😊🚀

**Comment by yaneony** [comment]
I did that with rethinkdb-ts driver now by extension. Added simple function
which do:
- read all indexes
- drop table
- create new table
- fill table with indexes above

Might be dirty hack, but working much faster than deleting over million
rows with .delete() command.

сб, 1 февр. 2020 г., 15:39 Boros Gábor <notifications@github.com>:
